prow/jira/fakejira/fake_test.go

```diff
@@ -72,3 +72,37 @@ func TestFakeClient_SearchWithContext(t *testing.T) {
 		t.Fatal("expected invalid query to fail, but got no error")
 	}
 }
+
+func TestFakeClient_GetIssueWithOptions(t *testing.T) {
+	responses := make(map[GetIssueRequest]GetIssueResponse)
+
+	issue := &jira.Issue{
+		ID:     "1234",
+		Fields: &jira.IssueFields{Project: jira.Project{Name: "test"}},
+	}
+
+	options := &jira.GetQueryOptions{Fields: "project"}
+
+	responses[GetIssueRequest{issueID: "1234", options: options}] = GetIssueResponse{
+		issue: issue,
+		error: nil,
+	}
+	fakeClient := &FakeClient{GetIssueResponses: responses}
+
+	r, err := fakeClient.GetIssueWithOptions("1234", options)
```

Comment:
We should pass `&jira.GetQueryOptions{Fields: "project"}` to this function call directly instead of reusing the same pointer we used to create the `GetIssueResponses` map. Reusing this pointer is the reason the issue I mentioned above was not caught by this unit test.

Reply:
Fixed